packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      checkListItemExistence(listTodosResult2, `list${modelName}`, todo['customId'], true, 'customId');
+
+      // owner(user1) can listen to updates on allowed fields
+      const todoRandom = {
+        ...todo,
+        customId: Date.now().toString(),
+      };
+      const todoRandomUpdated = {
+        customId: todoRandom.customId,
+        authors: [userName1],
+        privateContent: 'Private Content updated',
+      };
+      const subscriberClient = getConfiguredAppsyncClientCognitoAuth(graphQlEndpoint, region, userMap[userName1]);
+      const subTodoHelper = createModelOperationHelpers(subscriberClient, schema)[modelName];
+
+      const onCreateSubscriptionResult = await subTodoHelper.subscribe(
+        'onCreate',
+        [
+          async () => {
+            await user1TodoHelper.create(createResultSetName, todoRandom, completeOwnerResultSet);
+          },
+        ],
+        {},
+        completeOwnerResultSet,
+        false,
+      );
+      expect(onCreateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onCreateSubscriptionResult[0],
+        { ...todoRandom, privateContent: null, ownersContent: null },
+        `onCreate${modelName}`,
+      );
+
+      const onUpdateSubscriptionResult = await subTodoHelper.subscribe(
+        'onUpdate',
+        [
+          async () => {
+            await user1TodoHelper.update(updateResultSetName, todoRandomUpdated, completeOwnerResultSet);
+          },
+        ],
+        {},
+        completeOwnerResultSet,
+        false,
+      );
+      expect(onUpdateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onUpdateSubscriptionResult[0],
+        { ...todoRandomUpdated, privateContent: null, ownersContent: null },
+        `onUpdate${modelName}`,
+      );
+    });
+
+    test('Custom list of owners model auth and restricted field operations', async () => {
+      const modelName = 'TodoCustomOwnersContentVarious';
+      const user1ModelOperationHelpers = createModelOperationHelpers(appSyncClients[userPoolProvider][userName1], schema);
+      const user2ModelOperationHelpers = createModelOperationHelpers(appSyncClients[userPoolProvider][userName2], schema);
+      const user1TodoHelper = user1ModelOperationHelpers[modelName];
+      const user2TodoHelper = user2ModelOperationHelpers[modelName];
+
+      const todo = {
+        customId: Date.now().toString(),
+        privateContent: 'Private Content',
+      };
+      const createResultSetName = `create${modelName}`;
+      const updateResultSetName = `update${modelName}`;
+      const deleteResultSetName = `delete${modelName}`;
+      const privateResultSet = `
+        customId
+        privateContent
+      `;
+      const ownerResultSet = `
+        ${privateResultSet}
+        authors
+      `;
+      const completeOwnerResultSet = `
+        ${ownerResultSet}
+        ownersContent
+      `;
+
+      /*
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket